pkg/util/common.go

```diff
@@ -1347,25 +1304,27 @@ func DeleteCustomResourceDefinition(apiExtensionClient *apiextensionsclient.Clie
 }
 
 // WaitUntilPodsAreReady will wait for the pods to be ready
-func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) (bool, error) {
-	// timer starts the timer for timeoutInSeconds. If the task doesn't completed, return error
-	timeout := time.NewTimer(time.Duration(timeoutInSeconds) * time.Second)
-	// ticker starts and execute the task for every n intervals
-	ticker := time.NewTicker(10 * time.Second)
-	for {
-		select {
-		case <-timeout.C:
-			ticker.Stop()
-			return false, fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
-		case <-ticker.C:
-			ready, err := IsPodReady(clientset, namespace, labelSelector)
-			if err != nil || ready {
-				timeout.Stop()
-				ticker.Stop()
-				return ready, err
-			}
+func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) error {
+	numOfTimesToCheck := int64(10)
+	var ticker *time.Ticker
+	if timeoutInSeconds < numOfTimesToCheck {
```

Comment:
This logic isn't correct.  If the timeoutInSeconds is less than numofTimesToCheck then you also need to set numOfTimesToCheck to 1.  Otherwise you will check 10 times with a delay of timeoutInSeconds instead of timeoutInSeconds being the upper bound.

I think a Timer and Ticker, like was there before, is easier to understand.  We can check at whatever interval we see fit via the Ticker, if Timeout fires then we should do a final check of the pods and if still not running exit with error.